account_invoice_split_invoice/tests/test_split_invoice.py

```diff
@@ -0,0 +1,138 @@
+        journal_obj = self.registry('account.journal')
+        journal_obj.write(self.cr, self.uid, 1, {
+            'update_posted': True,
+        })
+
+        partner = self.registry('res.partner')
+        partner_id = partner.create(self.cr, self.uid, {
+            'name': 'Kevin',
+
+        })
+
+        account_id = self.registry('account.account').search(
+            self.cr, self.uid, [('type', '=', 'receivable')])[0]
+        invoice_obj = self.registry('account.invoice')
+        invoice_id = invoice_obj.create(self.cr, self.uid, {
+            'name': invoice_name,
+            'account_id': account_id,
+            'currency_id': 1,
+            'journal_id': 1,
+            'partner_id': partner_id,
+            'date_invoice': datetime.today()
+        })
+
+        return invoice_id
+
+    def _create_invoice_line(self, description, amount):
+        invoice_line_obj = self.registry('account.invoice.line')
+        invoice_line_id = invoice_line_obj.create(self.cr, self.uid, {
+            'name': description,
+            'price_subtotal': amount,
+            'invoice_id': self.invoice_id,
+        })
+        return invoice_line_id
+
+    def _test_write_lines(self, invoice):
+        # Test if the invoice has been copied
+
+        original_amount = invoice.amount_total
+        original_name = invoice.name
+        original_partner = invoice.partner_id
+        original_invoice_date = invoice.date_invoice
+        original_account = invoice.account_id
+        original_journal = invoice.journal_id
+
+        wizard_obj = self.registry('account.invoice.split.wizard')
+
+        wizard_id = wizard_obj.create(
+            self.cr, self.uid, dict(), context={
+                'active_id': self.invoice_id})
+
+        wizard = wizard_obj.browse(self.cr, self.uid, wizard_id)
+
+        new_invoice_id = wizard_obj._write_lines(
+            self.cr, self.uid, wizard_id, 'invoice_line_ids', [
+                (1, self.invoice_line_id1, {'split': True})], "")
+
+        invoice_obj = self.registry('account.invoice')
+
+        new_invoice = invoice_obj.browse(self.cr, self.uid, new_invoice_id)
+
+        # Test if the lines have been exactly copied
+
+        self.assertEqual(wizard.invoice_id.id, self.invoice_id)
+        self.assertNotEqual(wizard.invoice_id.id, new_invoice_id)
+        self.assertEqual(wizard.invoice_id.name, original_name)
+        self.assertEqual(
+            wizard.invoice_id.partner_id, original_partner)
+        self.assertEqual(
+            wizard.invoice_id.date_invoice, original_invoice_date)
+        self.assertEqual(
+            wizard.invoice_id.account_id, original_account)
+        self.assertEqual(
+            wizard.invoice_id.journal_id, original_journal)
+        self.assertEqual(
+            original_amount,
+            wizard.invoice_id.amount_total + new_invoice.amount_total
+        )
+
+    def test_open_invoice(self):
+
+        # test for invoice in 'open' state
```

Comment:
Use docstrings (google for it) for method comments. Explain in this comment the test scenario (split an open invoice).